SU2_CFD/src/numerics_direct_turbulent.cpp

```diff
@@ -1087,27 +1067,20 @@ CUpwSca_TurbSST::CUpwSca_TurbSST(unsigned short val_nDim,
 CUpwSca_TurbSST::~CUpwSca_TurbSST(void) {
 }
 
-void CUpwSca_TurbSST::ExtraADPreaccIn() {
-
-  AD::SetPreaccIn(V_i, nDim+3);
-  AD::SetPreaccIn(V_j, nDim+3);
-  
-}
-
 void CUpwSca_TurbSST::FinishResidualCalc(su2double *val_residual,
                                                su2double **val_Jacobian_i,
                                                su2double **val_Jacobian_j,
                                                CConfig *config) {
   
-  val_residual[0] = a0*Density_i*TurbVar_i[0]+a1*Density_j*TurbVar_j[0];
-  val_residual[1] = a0*Density_i*TurbVar_i[1]+a1*Density_j*TurbVar_j[1];
```

Comment:
What's your rationale behind making these density changes?

I understand the problems with reconstruction and why the value of density is questionable.  But it seems to me like removing density is incorrect.  There's two reasons I think that:

1. The SST variables are updated by the `CVariable::AddConservativeSolution` call.  In that functions, the delta in the variables is assumed to be weighted by density, e.g. rho*k.  Based on that you can conclude that the equations are equations for the density-weighted variables, and the residuals should also be density weighted.
2. In the source term for the SST variables, (`CSourcePieceWise_TurbSST`) there are terms like (`2/3*rho*k*diverg*Volume`).  These residuals are density weighted. To be consistent, all the residuals should be density weighted.

For a converged, steady state solution, I think these changes will only change the conditioning.  Similarly, I expect no impact for a low-Mach test case.  Have you tested these changes on a case like the RAE 2822?

Reply:
(I've been looking at the code a bit more to get an answer).
The rationale was: I did not want to divide the mass flux by an average density which would then break that "summation = 0" property we can use to improve conditioning a bit. Following that logic I linearized the other terms w.r.t. k and w instead of rho k and rho w, I did not change the source residual only its Jacobian. But I forgot to see how the solution was being updated so yeah in the end it is wrong.
Seeing how SA models need a volumetric flux anyway I will probably revert those changes to SST.